### PR TITLE
Populate `backend.configuration_dict` from `target_info` endpoint

### PR DESCRIPTION
**Pull request by vtomole**
Part of https://github.com/Infleqtion/client-superstaq/issues/342

### CODE REVIEW

File: qiskit-superstaq/qiskit_superstaq/superstaq_provider.py [hunk 1]
```diff
@@ -462,17 +462,6 @@ def process_dfe(self, ids: List[str]) -> float:
         """
         return self._client.process_dfe(ids)
 
-    def target_info(self, target: str) -> Dict[str, Any]:
```
Comment (vtomole) [inline qiskit-superstaq/qiskit_superstaq/superstaq_provider.py:465]: This is deleted cause it's not how qiskit users are supposed to be getting target info. They should be using `backend.configuration`

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend_test.py [hunk 2]
```diff
@@ -9,21 +12,17 @@
 
 import qiskit_superstaq as qss
 
+if TYPE_CHECKING:
+    from qiskit_superstaq.conftest import MockSuperstaqProvider
 
-def test_default_options() -> None:
```
Comment (vtomole) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend_test.py:13]: Unnecessary test: Don't affect coverage with it's removal and doesn't look like a regression test.

**Comment by vtomole** [comment]
Notebooks check won't pass until a new version of server is released.

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 3]
```diff
@@ -35,21 +35,24 @@ def __init__(self, provider: qss.SuperstaqProvider, target: str) -> None:
...
             "backend_version": "n/a",
-            "n_qubits": -1,
-            "basis_gates": None,
+            "n_qubits": target_info["num_qubits"],
+            "basis_gates": target_info["native_gate_set"],
             "gates": [],
             "local": False,
             "simulator": False,
             "conditional": False,
             "open_pulse": False,
```
Comment (bharat-thotakura) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:49]: Should something like this also be added?
```suggestion
            "conditional": True if target_info["conditional"] else False ,
            "open_pulse": True if target_info["open_pulse"] else False ,
```

**Review by bharat-thotakura — COMMENTED** [review]
Partial review

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py (hunk 3 above)
Comment (paaige) [reply qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:49]: I think these checks are already taken care of when the target info is initialized in the client

Comment (vtomole) [reply qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:49]: Oh i see what's going on

@bharat-thotakura  So the reason to do something like this is when the keys the target info endpoint if returning is different from the ones that the configuration dict wants. For example `"n_qubits": target_info["num_qubits"],`

In the case of your suggestions, `target_info` endpoint already returns the `condition` and `open_pulse` keys which are then updated in `self.configuration_dict.update(target_info)`

**Comment by vtomole** [comment]
@bharat-thotakura I did need to apply your suggestion for the other keys though: https://github.com/Infleqtion/client-superstaq/issues/678. Just pushed a commit to handle undefined keys. Please take a look.

**Review by bharat-thotakura — COMMENTED** [review]
I tested the changes locally and it seems fine to me. Though, I did notice that `qss.SuperstaqProvider().get_backend(<target>).configuration_dict` yields a couple of duplicate information. Namely, `'backend_name'` <-> `'target'`  and `'n_qubits'` <-> `'num_qubits'` list the same thing. I think this is because of ` self.configuration_dict.update(target_info)`, so it may be worth dropping the duplicate key info.

**Comment by vtomole** [comment]
Good catch. I've removed them from the final dict.

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 4]
```diff
@@ -35,21 +35,26 @@ def __init__(self, provider: qss.SuperstaqProvider, target: str) -> None:
...
             "memory": False,
-            "max_shots": -1,
+            "max_shots": None,
             "coupling_map": None,
         }
+        target_info.pop("num_qubits", None)
+        target_info.pop("native_gate_set", None)
```
Comment (bharat-thotakura) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:55]: Should there also be: 
``` 
target_info.pop("target", None)
```
given that `"backend_name": target` is present in the dict, or do we later want to update and list both the 'superstaq name' and 'provider name' -- e.g. "qtm_h1-1_qpu" and "Quantinuum H1-1"

Comment (vtomole) [reply qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:55]: Oops. I forgot to pop this one. Done.

**Comment by bharat-thotakura** [comment]
@vtomole No other changes from me :+1:

**Pull request merged**